sympy/core/relational.py

```diff
@@ -52,6 +52,8 @@ def _canonical_coeff(rel):
     rel = rel.canonical
     if not rel.is_Relational or rel.rhs.is_Boolean:
         return rel  # Eq(x, True)
+    if rel.lhs.is_Boolean:
+        return rel.reversed  # e.g.: Eq(True, x) -> Eq(x, True)
```

Comment:
I think it would be better to check for `Expr` here rather than `Boolean`.

Reply:
Alright, I opened an alternative PR (to let these tests run to completion, and cleaner commit history upon merge):
gh-27541